File: activity/src/main/java/com/fragment/FragParam.java
package com.fragment;

import android.app.Fragment;
import android.content.Context;
import android.view.View;


/**
 * The type Frag param.
 */
public class FragParam {

    Context context;
    int replaceId,
            enter = 0,
            exit = 0,
            popEnter = 0,
            popExit = 0;
    Fragment fragment;
    String tag;
    boolean enableAnimation = false,
            isBackStack = false;
    View[] sharedElements = null;

}

File: activity/src/main/java/com/fragment/FragParam.kt
package com.fragment

import android.content.Context
import android.support.v4.app.Fragment
import android.view.View


/**
 * The type Frag param.
 */
class FragParam {

    internal var context: Context? = null
    internal var replaceId: Int = 0
    internal var enter = 0
    internal var exit = 0
    internal var popEnter = 0
    internal var popExit = 0
    internal var fragment: Fragment? = null
    internal var tag: String? = null
    internal var enableAnimation = false
    internal var isBackStack = false
    internal var sharedElements: Array<View> = emptyArray()

}
